### PR TITLE
Update VPA k8s dependencies to 1.18.3

### LINKED ISSUE #3219: VerticalPodAutoscaler not compatible with k8s 1.18.x libraries
We are trying to vendor in the latest Kubernetes client-libraries. We are blocked from doing so as we also depend on VPA as a library

### PR DESCRIPTION
**Pull request by bskiba**
Fixes #3219 #3097 

Does similar work as #3098 but that PR has merge conflicts. I also updated the e2e test dependencies accordingly.

### CODE REVIEW

**Comment by bskiba** [comment]
Unrelated test failing.

**Comment by bskiba** [comment]
@krzysied would you be able to review?
These are all mechanical changes, no actual behavioral changes. I tried to split them into commits that make sense. Let me know if you would like me to split this in more PRs.

**Comment by bskiba** [comment]
Actually, I would like to test this a little more, you can hold off the review for now.

**Comment by bskiba** [comment]
Okay, this is ready for review
/assign @krzysied 
Krzysztof, would you mind taking a look? As mentioned before, this only updates dependencies and generated code and adjusts code to work with new deps.

File: vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go [hunk 1]
```diff
@@ -321,7 +316,7 @@ func (rc *ResourceConsumer) sendConsumeCustomMetric(delta int) {
 	framework.ExpectNoError(err)
 }
 
-// CleanUp func
+// CleanUp clean up the background goroutines responsible for consuming resources.
```
Comment (krzysied) [inline vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go:319]: s/clean/cleans/

File: vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go [hunk 2]
```diff
@@ -65,18 +65,16 @@ const (
...
 
 var (
-	resourceConsumerImage           = imageutils.GetE2EImage(imageutils.ResourceConsumer)
-	resourceConsumerControllerImage = imageutils.GetE2EImage(imageutils.ResourceController)
+	resourceConsumerImage = imageutils.GetE2EImage(imageutils.ResourceConsumer)
 )
 
 var (
-	// KindRC var
+	// KindRC is the GVK for ReplicationController
```
Comment (krzysied) [inline vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go:72]: nit: missing dot at the end of the sentence. The same goes for other comments.

Comment (bskiba) [reply vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go:72]: This file is mostly copied from k8s repo. I'm not entirely sure we need this copy, but I'd rather not mix it with this PR. The same goes for the other comment.

**Comment by bskiba** [comment]
/hold
until e2e tests are fixed

File: vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go (hunk 2 above)
Comment (krzysied) [reply vertical-pod-autoscaler/e2e/v1/autoscaling_utils.go:72]: ack.

**Comment by tariq1890** [comment]
Any updates on this?

**Comment by bskiba** [comment]
This is on hold until e2e tests are fixed. #3340 which should fix them has been merged just now, so we should see green test runs today. Then this change will be merged.

**Comment by bskiba** [comment]
Tests are green again \o/. 
/hold cancel

**Pull request merged**